### PR TITLE
Fix bug that can cause param/query labels to be dropped.

### PR DESCRIPTION
**Pull request by michaelochurch**

### CODE REVIEW

File: R/utils.r [hunk 1]
```diff
@@ -4,8 +4,26 @@ from_json <- function(obj) {
   simplify_homogeneous_lists(jsonlite::fromJSON(obj, simplifyVector = FALSE))
 }
 
+has_names <- function(obj) {
+  !is.null(names(obj)) && !is.null(Find(function(s) !is.na(s) && (s != ""), names(obj)))
```
Comment (robertzk) [inline R/utils.r]: I don't believe it's possible for names to be all blank in practice, so `!is.null(names(obj))` should be sufficient.

**Comment by robertzk** [comment]
Looks good, but verify my comment about `has_names`.

**Comment by michaelochurch** [comment]
It's pathological but it can happen.

```R
> bad <- c(1, 2, 3, 4)
> names(bad) <- c("")
> names(bad)
[1] "" NA NA NA
> is.null(names(bad))
[1] FALSE
> names(bad)[1] <- NA
> names(bad)
[1] NA NA NA NA
> is.null(names(bad))
[1] FALSE
```

**Comment by robertzk** [comment]
Nice example.

**Pull request merged**